src/clj/infinitelives/pixi/macros.clj

```diff
@@ -111,3 +112,33 @@
              (when ~form
                  ~@body
                  (recur ~test)))))
+
+(defmacro <!* [test & body]
+  `(let [result# (cljs.core.async/<! ~@body)]
+     (if ~test
+       result#
+       (throw (js/Error "go-while exit")))))
+
+(defmacro >!* [test & body]
+  `(let [result# (cljs.core.async/>! ~@body)]
+     (if ~test
+       result#
+       (throw (js/Error "go-while exit")))))
+
+(defn process-body [test [head & tail]]
+  (let [[processed-head processed-tail]
+          (cond
+            (identical? '() head) ['() tail]
+            (identical? '[] head) ['[] tail]
+            (list? head) [(process-body test head) tail]
+            (vector? head) [(vec (process-body test head)) tail]
+            (map? head) [(into {} (process-body test (seq head))) tail]
+            (= '<! head) ['baller.async/<!* (cons test tail)]
+            (= '>! head) ['baller.async/>!* (cons test tail)]
```

Comment:
This isn't going to work. Namespaced to baller.async instead of infinitelives.pixi.macros.

Anyway...

I don't want to bring in this async stuff until it works right. Presently the macro breaks inner empty parameter (loop [] ... (recur)) calls, and breaks (set! (.-foo obj) ...) calls.

Can we keep these in the individual projects for now?